components/brave_wallet/browser/simple_hash_client.cc

```diff
@@ -591,6 +898,289 @@ SimpleHashClient::ParseNFTsFromSimpleHash(const base::Value& json_value,
   return std::make_pair(next_cursor, std::move(nft_tokens));
 }
 
+std::optional<SolCompressedNftProofData>
+SimpleHashClient::ParseSolCompressedNftProofData(
+    const base::Value& json_value) {
+  const base::Value::Dict* dict = json_value.GetIfDict();
+  if (!dict) {
+    return std::nullopt;
+  }
+
+  SolCompressedNftProofData result;
+
+  const std::string* root_opt = dict->FindString("root");
+  const std::string* data_hash_opt = dict->FindString("data_hash");
+  const std::string* creator_hash_opt = dict->FindString("creator_hash");
+  uint64_t leaf_index = 0;
+  if (!GetUint64FromDictValue(*dict, "leaf_index", false, &leaf_index)) {
+    return std::nullopt;
+  }
+  const std::string* owner_opt = dict->FindString("owner");
+  const std::string* merkle_tree_opt = dict->FindString("merkle_tree");
+  const std::string* delegate_opt = dict->FindString("delegate");
+  uint64_t canopy_depth = 0;
+  if (!GetUint64FromDictValue(*dict, "canopy_depth", false, &canopy_depth)) {
```

Comment:
Also check against UINT32_MAX so we won't accidentally get a value bigger than uint32 then downcast it to uint32.